### ISSUE TITLE
🚀 🚀 🚀 Test automatici sui dati (please!)

### ISSUE THREAD

**Issue by floatingpurr** [pcm-dpc/COVID-19#945]
Ciao @umbros, ti taggo direttamente in qualità di responsabile tecnico di questo repository. Mi domandavo se fosse possibile implementare una serie di test automatici per evitare i soliti problemi ricorrenti sui dati come: #943, #923, #911, #908 etc...

In questi casi non si tratta di entrare nel merito di un numero da certificare, ma 
Se vuoi vedere un esempio,

**Comment by miccoli** [comment]
Tra le issue che è singolare non siano ancora state risolte c'è la #863. Qualsiasi controllo automatico dei dati può funzionare solo se si parte da una serie di ipotesi su come i dati devono essere organizzati e quali invariati devono valere, altrimenti cosa controllare?

Certo che il minimo sindacale sarebbe almeno il parsing corretto dei file .json e .csv...

Comunque se ci fossero indicazioni precise da parte dei responsabili sono ovviamente interessato ad implementarle nella mia #847 

Come già detto in altro contesto, la procedura corretta, una volta implementate le GitHub actions di QA sarebbe 
- creare la branch YYYYMMDD con I file del giorno
- fare il push della branch su questa repo
- verificare che tutto sia OK, ed eventualmente iterare fino a che la procedura di validazione dà luce verde
- fare un merge su master
- cancellare la branch YYYYMMDD

**Comment by floatingpurr** [comment]
Concordo, le ipotesi di partenza sono fondamentali per ogni strategia di verifica e di QA. Non volevo essere troppo generico. Mi immaginavo semplicemente una collezione di step di verifica modulari che partano dal check della mera serializzazione del dato, per arrivare alla sua consistenza e validità semantica.

Mi rendo conto che più si sale di livello, più le correzioni necessito di processi di certificazione del numero in sé che trascendono l'aspetto tecnico. Però dai, siamo onesti: l'unexpected token, l'encoding, il dato duplicato, sono dei problemi squisitamente tecnici che possono essere scovati automaticamente e corretti prima del rilascio sul master evitando non pochi problemi. Per esempio, in questo momento, chi consuma i dati via JSON è bloccato da quasi 24h a causa della #943 / #941 e deve barcamenarsi con workaround manuali.

Ottimo approccio la procedura di Actions descritta su. @umbros che ne dici, se ne può parlare?

**Comment by alexodus** [comment]
Anche oggi ci sono errori di formattazione.

**Comment by floatingpurr** [comment]
Ref. #948

**Comment by floatingpurr** [comment]
Ritorno sul tema dopo un paio di settimane di attesa di un feedback ritaggando @umbros. Volevo soltanto capire se ci fosse un'apertura verso le proposte descritte in questa issue, oppure no, così da capire come orientarsi.

Grazie.

**Comment by floatingpurr** [comment]
Ciao @umbros, scusami se ti ritaggo ma forse questa issue ti era sfuggita. Vorrei solo sapere se puoi darci gentilmente un feedback riguardo a questa richiesta e a questo genere di problemi.

Grazie!
